day_adding/calendar.rb

```diff
@@ -26,15 +14,34 @@ def add_one_day(year, month, day)
     12 => 31
   }
 
-  if month == 12 && day == days_in_month[month]
-    return "#{year+1}-1-1"
-  elsif month == 2 && day == 28 && year_leap?(year)
-    return "#{year}-#{month}-#{day+1}"
-  elsif month == 2 && day == 29 && year_leap?(year)
-    return "#{year}-#{month+1}-1"
-  elsif day == days_in_month[month]
-    return "#{year}-#{month+1}-1"
-  else 
-    return "#{year}-#{month}-#{day+1}"
+  def self.year_leap?(year)
+    if (year % 400).zero?
+      true
+    elsif (year % 100).zero?
+      false
+    elsif (year % 4).zero?
+      true
+    else
+      false
+    end
+  end
+
+  def self.add_one_day(year, month, day)
+    day_count = DAYS_IN_MONTH[month]
+
+    case
+    when month == 2 && year_leap?(year)
+      if day == 28
```

Comment:
`if` inside `case` doesn't look very well. I think it apparently would be better to use `if` form here. 

However, you've learned pretty well how `case` works 👍